extensions/mailer/runtime/src/main/java/io/quarkus/mailer/MailTemplate.java

```diff
@@ -0,0 +1,57 @@
+package io.quarkus.mailer;
+
+import java.util.concurrent.CompletionStage;
+
+/**
+ * Represents an e-mail definition based on a template.
+ */
+public interface MailTemplate {
+
+    /**
+     * 
+     * @return a new template instance
+     */
+    MailTemplateInstance instance();
+
+    default MailTemplateInstance of(Mail mail) {
+        return instance().mail(mail);
+    }
+
+    default MailTemplateInstance to(String... values) {
+        return instance().to(values);
+    }
+
+    default MailTemplateInstance data(String key, Object value) {
+        return instance().data(key, value);
+    }
+
+    /**
+     * Represents an instance of {@link MailTemplate}.
+     * <p>
+     * This construct is not thread-safe.
+     */
+    interface MailTemplateInstance {
+
+        MailTemplateInstance mail(Mail mail);
+
+        MailTemplateInstance to(String... to);
+
+        MailTemplateInstance cc(String... cc);
+
+        MailTemplateInstance bcc(String... bcc);
+
+        MailTemplateInstance subject(String subject);
+
+        MailTemplateInstance from(String from);
+
+        MailTemplateInstance replyTo(String replyTo);
+
+        MailTemplateInstance bounceAddress(String bounceAddress);
+
+        MailTemplateInstance data(String key, Object value);
```

Comment:
Let's also plan to include the `Locale` at some point.

Reply:
Yes, that's definitely on the plan... once we support `Locale` in templates in general ;-).